test/test-span-data.ts

```diff
@@ -14,177 +14,174 @@
-
-function createRootSpanData(name, traceId?, parentId?, skipFrames?) {
-  return new RootSpanData(
-    { projectId: '', traceId: traceId, spans: [] },
-    name,
-    parentId,
-    skipFrames
-  );
-}
-
-describe('SpanData', function() {
-  var agent;
-  before(function() {
-    agent = require('../..').start({
-      projectId: '0',
-      samplingRate: 0
-    });
+import {traceWriter, TraceWriter, TraceWriterConfig} from '../src/trace-writer';
+
+import {TestLogger} from './logger';
+import * as traceAgentModule from './trace';
+
+describe('SpanData', () => {
+  class CaptureSpanTraceWriter extends TraceWriter {
+    writeSpan(trace: Trace) {
+      capturedTrace = trace;
+    }
+  }
+  let capturedTrace: Trace;
+  let trace: Trace;
+
+  before(() => {
+    traceAgentModule.setTraceWriter(CaptureSpanTraceWriter);
+    traceWriter.create(
+        {
+          onUncaughtException: 'ignore',
+          maximumLabelValueSize: 16,
+          stackTraceLimit: 5
+        } as TraceWriterConfig,
+        new TestLogger());
   });
 
-  it('has correct default values', function() {
-    cls.get().runWithNewContext(function() {
-      var spanData = createRootSpanData('name', 1, 2);
-      assert.ok(spanData.trace);
-      assert.strictEqual(spanData.trace.traceId, 1);
-      assert.ok(spanData.span.spanId);
-      assert.strictEqual(spanData.span.name, 'name');
-    });
+  after(() => {
+    traceAgentModule.setTraceWriter(traceAgentModule.TestTraceWriter);
   });
 
-  it('generates unique numeric span ID strings', function() {
-    cls.get().runWithNewContext(function() {
-      var spans = [];
-      var spanIds = new Set<string>();
-      var rootSpanData = createRootSpanData('hi');
-      var numSpanIdsToCheck = 5;
-      for (var i = 0; i < numSpanIdsToCheck; i++) {
-        var spanData = new ChildSpanData(
-            rootSpanData.trace, 'child', rootSpanData.span.spanId, 0);
-        var spanId = spanData.span.spanId;
-        spanIds.add(spanId);
-        // Check that the span IDs are numeric positive number strings
-        assert.ok(typeof spanId === 'string');
-        assert.ok(spanId.match(/\d+/));
-        assert.ok(Number(spanId) > 0);
-        assert.strictEqual(Number(spanId).toString(), spanId);
-      }
-    });
+  beforeEach(() => {
+    trace = {projectId: '0', traceId: 'trace-id', spans: []};
   });
 
-  it('converts label values to strings', function() {
-    cls.get().runWithNewContext(function() {
-      var spanData = createRootSpanData('name', 1, 2);
-      spanData.addLabel('a', 'b');
-      assert.strictEqual(spanData.span.labels.a, 'b');
-      spanData.addLabel('c', 5);
-      assert.strictEqual(spanData.span.labels.c, '5');
+  describe('BaseSpanData', () => {
+    // We use the ChildSpanData constructor, but what we are really testing
+    // is the API that is common between child and root spans -- which is
+    // BaseSpanData.
+    class CommonSpanData extends ChildSpanData {}
```

Comment:
Since we test functionality common to `Child` and `RootSpanData`, I'd rather the constructor not specifically say `ChildSpanData`. That's the only reason.